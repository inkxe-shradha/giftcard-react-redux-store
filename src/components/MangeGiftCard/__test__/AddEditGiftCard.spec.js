import { ThemeProvider } from "@mui/material";
import { createTheme } from "@mui/material";
import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import { mount } from "enzyme";
import AddEditGiftCard from "../AddEditGiftCard";
const theme = createTheme();

describe("Gift card Manage Test Suite", () => {
  const pageProps = {
    isOpen: true,
    onClose: jest.fn(),
    handelSubmitted: jest.fn(),
    errorMessage: "",
    isLoading: false,
    giftCard: {},
    mode: "add",
  };
  const customRender = (props = pageProps, type = "render") => {
    if (type === "render") {
      return render(
        <ThemeProvider theme={theme}>
          <AddEditGiftCard {...props} />
        </ThemeProvider>
      );
    } else {
      return mount(
        <ThemeProvider theme={theme}>
          <AddEditGiftCard {...props} />
        </ThemeProvider>
      );
    }
  };

  it("Should render the component snapshot", () => {
    const view = customRender(pageProps, "mount");
    expect(view.html()).toMatchSnapshot();
  });

  it("Should have the title send mail in the modal", () => {
    customRender();
    expect(screen.getAllByText("Send Email")).toHaveLength(2);
  });

  it("Should be contain all the input field with valid name", async () => {
    customRender();
    expect(screen.getByLabelText(/Card Points/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/Card Name/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/Card Category Name/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/Card Retailer/i)).toBeInTheDocument();
    expect(
      screen.getByLabelText(/Card Short Descriptions/i)
    ).toBeInTheDocument();
    expect(
      screen.getByLabelText(/Card Long Descriptions/i)
    ).toBeInTheDocument();
    expect(screen.getByLabelText(/Card Image Url/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/Card Vendor/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/Number of card count/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/Expires On/i)).toBeInTheDocument();
  });

  it("Should be empty field  when the form loaded", () => {
    customRender();
    const submitButton = screen.getByRole("button", { name: "Save" });
    const form = screen.getAllByTestId("add-edit-gift-card-form")[0];
    fireEvent.click(submitButton);
    expect(form).toHaveFormValues({
      cardName: "",
      cardCategory: "",
      cardRetailer: "",
      cardShortDesc: "",
      cardLongDesc: "",
      cardImage: "",
      cardVendor: "",
      cardCount: null,
      cardExpiryDate: "",
    });
  });

  it("Should be validate the form field once the submit button is clicked", async () => {
    customRender();
    const submitButton = screen.getByRole("button", { name: "Save" });
    fireEvent.click(submitButton);
    await waitFor(() => {
      const errorStatement = screen.getByText(/Card name is required/i);
      expect(errorStatement).toBeInTheDocument();
    });
    await waitFor(() => {
      const errorStatement = screen.getByText(/Card points is required/i);
      expect(errorStatement).toBeInTheDocument();
    });
    await waitFor(() => {
      const errorStatement = screen.getByText(/Card category is required/i);
      expect(errorStatement).toBeInTheDocument();
    });
    await waitFor(() => {
      const errorStatement = screen.getByText(/Card retailer is required/i);
      expect(errorStatement).toBeInTheDocument();
    });
    await waitFor(() => {
      const errorStatement = screen.getByText(/Card expiry date is required/i);
      expect(errorStatement).toBeInTheDocument();
    });
    await waitFor(() => {
      const errorStatement = screen.getByText(/Card image is required/i);
      expect(errorStatement).toBeInTheDocument();
    });
    await waitFor(() => {
      const errorStatement = screen.getByText(/Card vendor is required/i);
      expect(errorStatement).toBeInTheDocument();
    });
    await waitFor(() => {
      const errorStatement = screen.getByText(/Card description is required/i);
      expect(errorStatement).toBeInTheDocument();
    });
  });

  it("Should show the same form value as it pass down from component props in giftcard Obj", async () => {
    const onSubmit = jest.fn();
    const giftCard = {
      cardName: "Test Card",
      cardCategory: "Test Category",
      cardRetailer: "Test Retailer",
      cardShortDesc: "Test Short Desc",
      cardLongDesc: "Test Long Desc",
      cardImage: "Test Image",
      cardVendor: "Test Vendor",
      cardCount: 1,
      cardExpiryDate: "2020-01-01",
    };
    const pageProps = {
      isOpen: true,
      onClose: jest.fn(),
      handelSubmitted: jest.fn(),
      errorMessage: "",
      isLoading: false,
      giftCard,
      mode: "edit",
    };
    customRender(pageProps);
    const submitButton = screen.getByRole("button", { name: "Save" });
    fireEvent.click(submitButton);
    await waitFor(() => {
      const form = screen.getAllByTestId("add-edit-gift-card-form")[1];
      expect(form).toHaveFormValues({
        cardName: "Test Card",
        cardCategory: "Test Category",
        cardRetailer: "Test Retailer",
        cardShortDesc: "Test Short Desc",
        cardLongDesc: "Test Long Desc",
        cardImage: "Test Image",
        cardVendor: "Test Vendor",
        cardCount: 1,
        cardExpiryDate: "2020-01-01",
      });
    });
  });

  it("Should called the onClose function when the  card is closed", () => {
    customRender();
    const closeButton = screen.getAllByLabelText(/Close/i)[0];
    fireEvent.click(closeButton);
    expect(pageProps.onClose).toHaveBeenCalled();
  });

  it("Should called the handelSubmitted function when the  card is submitted", async () => {
    const giftCard = {
      cardName: "Uber Gift Card",
      cardPoints: 12,
      cardCategory: "Ecommerce",
      cardRetailer: "Amazon",
      cardIssueDate: "2022-04-18T07:11:54.233Z",
      cardExpiryDate: "2025-01-01T00:00:00.000Z",
      cardCount: 91,
      cardImage: "https://source.unsplash.com/random/250*250?apple",
      cardVendor: "Uber",
      cardShortDesc: "10% OFF",
      cardLongDesc:
        " Lorem ipsum dolor sit amet consectetur adipisicing elit. Molestiae quisquam tenetur alias ducimus nostrum voluptate, nam suscipit ipsa dolorem mollitia error soluta, distinctio esse. Earum, quo! Cupiditate dignissimos iste ratione.",
      cardComments: [],
      id: 1650265917967,
    };
    const pageProps = {
      isOpen: true,
      onClose: jest.fn(),
      handelSubmitted: jest.fn(),
      errorMessage: "",
      isLoading: false,
      giftCard,
      mode: "edit",
    };
    customRender(pageProps);
    const submitButton = screen.getByRole("button", { name: "Save" });
    fireEvent.click(submitButton);
    await waitFor(() => {
      expect(pageProps.handelSubmitted).toHaveBeenCalled();
    });
  });
});
