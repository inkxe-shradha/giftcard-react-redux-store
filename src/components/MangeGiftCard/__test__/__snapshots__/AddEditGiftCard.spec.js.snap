// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift card Manage Test Suite Should render the component snapshot 1`] = `"<div role=\\"presentation\\" aria-labelledby=\\"Modal for opening the Add Edit Giftcard modal\\" aria-describedby=\\"Giftcard modal opening it.\\" class=\\"MuiModal-root css-79ws1d-MuiModal-root\\"><div aria-hidden=\\"true\\" class=\\"MuiBackdrop-root css-i9fmh8-MuiBackdrop-root-MuiModal-backdrop\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div><div tabindex=\\"0\\" data-test=\\"sentinelStart\\"></div><div class=\\"MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root\\" tabindex=\\"-1\\"><div class=\\"makeStyles-modalBox-1 MuiBox-root css-1cvh3gv\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium makeStyles-iconButton-2 css-78trlr-MuiButtonBase-root-MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"close\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"CloseIcon\\"><path d=\\"M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z\\"></path></svg><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button><h2 class=\\"MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root\\" id=\\"modal-modal-title\\">Send Email</h2><form data-testid=\\"add-edit-gift-card-form\\" class=\\"row\\" novalidate=\\"\\"><div class=\\"row mb-1\\"><div class=\\"col-md-6\\"><div class=\\"MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root\\"><label class=\\"MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root\\" data-shrink=\\"false\\" for=\\"cardName\\" id=\\"cardName-label\\">Card Name<span aria-hidden=\\"true\\" class=\\"MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk\\"> *</span></label><div class=\\"MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root\\"><input aria-invalid=\\"false\\" id=\\"cardName\\" name=\\"cardName\\" required=\\"\\" type=\\"text\\" class=\\"MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input\\" value=\\"\\"><fieldset aria-hidden=\\"true\\" class=\\"MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline\\"><legend class=\\"css-1ftyaf0\\"><span>Card Name&nbsp;*</span></legend></fieldset></div></div></div><div class=\\"col-md-6\\"><div class=\\"MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root\\"><label class=\\"MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root\\" data-shrink=\\"false\\" for=\\"cardPoints\\" id=\\"cardPoints-label\\">Card Points<span aria-hidden=\\"true\\" class=\\"MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk\\"> *</span></label><div class=\\"MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root\\"><input aria-invalid=\\"false\\" id=\\"cardPoints\\" name=\\"cardPoints\\" required=\\"\\" type=\\"number\\" class=\\"MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input\\" value=\\"\\"><fieldset aria-hidden=\\"true\\" class=\\"MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline\\"><legend class=\\"css-1ftyaf0\\"><span>Card Points&nbsp;*</span></legend></fieldset></div></div></div></div><div class=\\"row mb-1\\"><div class=\\"col-md-6\\"><div class=\\"MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root\\"><label class=\\"MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root\\" data-shrink=\\"false\\" for=\\"cardCategory\\" id=\\"cardCategory-label\\">Card Category Name<span aria-hidden=\\"true\\" class=\\"MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk\\"> *</span></label><div class=\\"MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root\\"><input aria-invalid=\\"false\\" id=\\"cardCategory\\" name=\\"cardCategory\\" required=\\"\\" type=\\"text\\" class=\\"MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input\\" value=\\"\\"><fieldset aria-hidden=\\"true\\" class=\\"MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline\\"><legend class=\\"css-1ftyaf0\\"><span>Card Category Name&nbsp;*</span></legend></fieldset></div></div></div><div class=\\"col-md-6\\"><div class=\\"MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root\\"><label class=\\"MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root\\" data-shrink=\\"false\\" for=\\"cardRetailer\\" id=\\"cardRetailer-label\\">Card Retailer<span aria-hidden=\\"true\\" class=\\"MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk\\"> *</span></label><div class=\\"MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root\\"><input aria-invalid=\\"false\\" id=\\"cardRetailer\\" name=\\"cardRetailer\\" required=\\"\\" type=\\"text\\" class=\\"MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input\\" value=\\"\\"><fieldset aria-hidden=\\"true\\" class=\\"MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline\\"><legend class=\\"css-1ftyaf0\\"><span>Card Retailer&nbsp;*</span></legend></fieldset></div></div></div></div><div class=\\"row mb-1\\"><div class=\\"col-md-8\\"><div class=\\"MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root\\"><label class=\\"MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root\\" data-shrink=\\"false\\" for=\\"cardImage\\" id=\\"cardImage-label\\">Card Image Url<span aria-hidden=\\"true\\" class=\\"MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk\\"> *</span></label><div class=\\"MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root\\"><input aria-invalid=\\"false\\" id=\\"cardImage\\" name=\\"cardImage\\" required=\\"\\" type=\\"url\\" class=\\"MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input\\" value=\\"\\"><fieldset aria-hidden=\\"true\\" class=\\"MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline\\"><legend class=\\"css-1ftyaf0\\"><span>Card Image Url&nbsp;*</span></legend></fieldset></div></div></div><div class=\\"col-md-4\\"><div class=\\"MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root\\"><label class=\\"MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root\\" data-shrink=\\"false\\" for=\\"cardCount\\" id=\\"cardCount-label\\">Number of card count<span aria-hidden=\\"true\\" class=\\"MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk\\"> *</span></label><div class=\\"MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root\\"><input aria-invalid=\\"false\\" id=\\"cardCount\\" name=\\"cardCount\\" required=\\"\\" type=\\"number\\" class=\\"MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input\\" value=\\"\\"><fieldset aria-hidden=\\"true\\" class=\\"MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline\\"><legend class=\\"css-1ftyaf0\\"><span>Number of card count&nbsp;*</span></legend></fieldset></div></div></div></div><div class=\\"row mb-1\\"><div class=\\"col-md-4\\"><div class=\\"MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root\\"><label class=\\"MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root\\" data-shrink=\\"false\\" for=\\"cardVendor\\" id=\\"cardVendor-label\\">Card Vendor<span aria-hidden=\\"true\\" class=\\"MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk\\"> *</span></label><div class=\\"MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root\\"><input aria-invalid=\\"false\\" id=\\"cardVendor\\" name=\\"cardVendor\\" required=\\"\\" type=\\"text\\" class=\\"MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input\\" value=\\"\\"><fieldset aria-hidden=\\"true\\" class=\\"MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline\\"><legend class=\\"css-1ftyaf0\\"><span>Card Vendor&nbsp;*</span></legend></fieldset></div></div></div><div class=\\"col-md-4\\"><div class=\\"MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root\\"><label class=\\"MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root\\" data-shrink=\\"false\\" for=\\"cardShortDesc\\" id=\\"cardShortDesc-label\\">Card Short Descriptions<span aria-hidden=\\"true\\" class=\\"MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk\\"> *</span></label><div class=\\"MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root\\"><input aria-invalid=\\"false\\" id=\\"cardShortDesc\\" name=\\"cardShortDesc\\" required=\\"\\" type=\\"text\\" class=\\"MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input\\" value=\\"\\"><fieldset aria-hidden=\\"true\\" class=\\"MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline\\"><legend class=\\"css-1ftyaf0\\"><span>Card Short Descriptions&nbsp;*</span></legend></fieldset></div></div></div><div class=\\"col-md-4\\"><div class=\\"MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root\\"><label class=\\"MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root\\" data-shrink=\\"false\\" for=\\"cardExpiryDate\\" id=\\"cardExpiryDate-label\\">Expires On<span aria-hidden=\\"true\\" class=\\"MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk\\"> *</span></label><div class=\\"MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root\\"><input aria-invalid=\\"false\\" id=\\"cardExpiryDate\\" name=\\"cardExpiryDate\\" required=\\"\\" type=\\"date\\" class=\\"MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input\\" value=\\"\\"><fieldset aria-hidden=\\"true\\" class=\\"MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline\\"><legend class=\\"css-1ftyaf0\\"><span>Expires On&nbsp;*</span></legend></fieldset></div></div></div></div><div class=\\"row mb-1\\"><div class=\\"col-md-8 offset-md-2\\"><div class=\\"MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root\\"><label class=\\"MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root\\" data-shrink=\\"false\\" for=\\"cardLongDesc\\" id=\\"cardLongDesc-label\\">Card Long Descriptions<span aria-hidden=\\"true\\" class=\\"MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk\\"> *</span></label><div class=\\"MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline css-8ewcdo-MuiInputBase-root-MuiOutlinedInput-root\\"><textarea rows=\\"4\\" aria-invalid=\\"false\\" id=\\"cardLongDesc\\" name=\\"cardLongDesc\\" required=\\"\\" class=\\"MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input\\" style=\\"height: 0px; overflow: hidden;\\"></textarea><textarea aria-hidden=\\"true\\" class=\\"MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input\\" readonly=\\"\\" tabindex=\\"-1\\" style=\\"visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;\\"></textarea><fieldset aria-hidden=\\"true\\" class=\\"MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline\\"><legend class=\\"css-1ftyaf0\\"><span>Card Long Descriptions&nbsp;*</span></legend></fieldset></div></div></div></div><div class=\\"row mt-2\\"><div class=\\"col-md-12 text-center\\"><button class=\\"MuiLoadingButton-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-wiajk6-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root\\" tabindex=\\"0\\" type=\\"submit\\" id=\\":ra:\\" margin=\\"normal\\">Save<span class=\\"MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"SendIcon\\"><path d=\\"M2.01 21 23 12 2.01 3 2 10l15 2-15 2z\\"></path></svg></span><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button><button class=\\"MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-1sszw6i-MuiButtonBase-root-MuiButton-root\\" tabindex=\\"0\\" type=\\"reset\\" margin=\\"normal\\"><span class=\\"MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"RestartAltIcon\\"><path d=\\"M12 5V2L8 6l4 4V7c3.31 0 6 2.69 6 6 0 2.97-2.17 5.43-5 5.91v2.02c3.95-.49 7-3.85 7-7.93 0-4.42-3.58-8-8-8zm-6 8c0-1.65.67-3.15 1.76-4.24L6.34 7.34C4.9 8.79 4 10.79 4 13c0 4.08 3.05 7.44 7 7.93v-2.02c-2.83-.48-5-2.94-5-5.91z\\"></path></svg></span>Reset<span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div></div></form></div></div><div tabindex=\\"0\\" data-test=\\"sentinelEnd\\"></div></div>"`;
